core/src/main/scala/com/yahoo/maha/core/RequestModel.scala

```diff
@@ -572,6 +572,14 @@ object RequestModel extends Logging {
               require(filterMap.contains(alias), s"Missing required filter: cube=${publicFact.name}, field=$alias")
           }
 
+          val optionalFilterColumns = publicFact.optionalFilterColumns
```

Comment:
Can we call this 'requiredFilterColumns' instead of 'optionalFilterColumns' and add a code comment in publicFact explaining that only one of them is required.